### PR TITLE
Retina images

### PR DESCRIPTION
**Pull request by jorilallo**
Implemented some PNG inspection code to determine if uploaded images are retina screenshots to scale them properly. This was hard enough so didn't bother to start looking at the blob stage size but that would be an improvement.

### CODE REVIEW

File: src/components/Image.js [hunk 1]
```diff
@@ -2,14 +2,64 @@
...
     hasError: false,
+    imageWidth: undefined,
+  };
+
+  /**
+   * Detect if PNG image is in retina resolution.
+   *
+   * PNG Chunk documentation: https://www.w3.org/TR/PNG-Chunks.html
+   */
+  onImgLoad = async (ev: SyntheticEvent<HTMLImageElement>) => {
```
Comment (tommoor) [inline src/components/Image.js]: I guess you have no desire to pull this into it's own module? I'd probably move into it's own lib file at least.

Will test thoroughly this w/e.

Comment (jorilallo) [reply src/components/Image.js]: Moved to a lib file. Not particularly fond of publishing 10 lines as a module :)

**Comment by tommoor** [comment]
Unfortunately the PHYs chunk seems to be missing when copying an image straight to your clipboard (hold shift while taking screenshot) 😢 – not sure why that would be, but saving to disk first works.

File: src/components/Image.js [hunk 2]
```diff
@@ -3,13 +3,28 @@ import * as React from "react";
...
   state = {
     hasError: false,
+    imageWidth: undefined,
+  };
+
+  onImgLoad = async (ev: SyntheticEvent<HTMLImageElement>) => {
+    const image = ev.target;
+    const url = this.props.node.data.get("src");
+    // Skip upload stage
+    if (url.startsWith("blob:")) return;
```
Comment (tommoor) [inline src/components/Image.js]: In my testing you don't need this line at all, the browser can fetch a blob uri without issue

**Comment by tommoor** [comment]
It seems like there's always a flash of the image at full size causing all the layout to shift, have you seen that in your testing?

**Pull request closed without merging**